In [1]:
import requests

/Users/salisfadi/Documents/calgary-transit-analytics/venv/lib/python3.9/site-packages/urllib3/__init__.py:35: NotOpenSSLWarning: urllib3 v2 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'LibreSSL 2.8.3'. See: https://github.com/urllib3/urllib3/issues/3020
  warnings.warn(


In [2]:
url = "https://data.calgary.ca/resource/ekzc-4ss8.json"
response = requests.get(url)

In [3]:
print(response.status_code)

403


In [4]:
print(response.text[:500])

{
  "error" : true,
  "message" : "You must be logged in to access this resource"
}



In [5]:
headers = {"User-Agent": "Mozilla/5.0"}
response2 = requests.get(url, headers=headers)
print(response2.status_code)

403


In [7]:
catalog_url = "https://data.calgary.ca/resource/3fy4-eztt.json"
params = {"$q": "fleet vehicle", "$limit": 20}
response3 = requests.get(catalog_url, params=params)
results = response3.json()

for item in results:
    print(item.get("name"), "→", item.get("uid"))

In [8]:
print(len(results))
print(results[:2])

0
[]


In [9]:
params = {"$q": "fleet", "$limit": 20}
response3 = requests.get(catalog_url, params=params)
results = response3.json()

print(len(results))
for item in results:
    print(item.get("name"), "→", item.get("uid"))

0


In [10]:
params = {"$q": "ridership", "$limit": 20}
response3 = requests.get(catalog_url, params=params)
results = response3.json()

print(len(results))
for item in results:
    print(item.get("name"), "→", item.get("uid"))

3
Monthly Ridership By Year → nypk-snzd
Calgary Transit Ridership → iema-jbc4
Yearly Ridership (current year is year-to-date) → n9it-gzsq


In [11]:
for term in ["vehicle", "bus", "transit fleet", "fleet list"]:
    params = {"$q": term, "$limit": 10}
    r = requests.get(catalog_url, params=params)
    items = r.json()
    print(f"\nSearch term: '{term}' → {len(items)} results")
    for item in items:
        print(" -", item.get("name"), "→", item.get("uid"))


Search term: 'vehicle' → 10 results
 - Traffic Volumes for 2018 → wwf6-cpsg
 - Traffic Counts at Permanent stations → vuyp-sbjp
 - Heat map - Based on all Traffic Incidents → yj8z-ptzu
 - Calgary Transit Realtime Trip Updates GTFS-RT → gs4m-mdc2
 - Calgary Transit Realtime Vehicle Positions GTFS-RT → am7c-qe3u
 - 2024 Traffic Volume Flow Map → u2zq-hjgc
 - Central Business District Cordon Count 2016 → dsje-mu8i
 - Snow Route Parking Bans → a84c-t6j3
 - Traffic Volumes for 2019 → qeqv-tb2c
 - Traffic Volumes for 2016 → 6wve-2ets

Search term: 'bus' → 10 results
 - Calgary Transit Stops By Route Cross Reference → pm3p-838w
 - Calgary Transit Stops → muzh-c9qc
 - Calgary Transit Realtime Trip Updates GTFS-RT → gs4m-mdc2
 - Calgary Transit Realtime Vehicle Positions GTFS-RT → am7c-qe3u
 - Calgary Transit Routes → hpnd-riq4
 - Hours of service provided by Calgary Transit → 64iy-rpqq
 - Annual Corporate Energy Use by Bus Units → msye-bzwu
 - Snow and Ice Clearing Priority Routes Map → fuea-

In [12]:
test_url = "https://data.calgary.ca/resource/msye-bzwu.json"
test_response = requests.get(test_url, params={"$limit": 5})
print(test_response.status_code)
print(test_response.text[:1000])

200
[{}
,{}
,{}
,{}
,{}]



In [13]:
metadata_url = "https://data.calgary.ca/api/views/msye-bzwu.json"
meta_response = requests.get(metadata_url)
print(meta_response.status_code)
meta = meta_response.json()
print(meta.get("name"))
print(meta.get("description"))
print([col["name"] for col in meta.get("columns", [])])

200
Annual Corporate Energy Use by Bus Units
Energy consumption at City facilities. This raw data is subject to periodic adjustments and modifications.
[]


In [14]:
metadata_url2 = "https://data.calgary.ca/api/views/am7c-qe3u.json"
meta_response2 = requests.get(metadata_url2)
print(meta_response2.status_code)
meta2 = meta_response2.json()
print(meta2.get("name"))
print(meta2.get("description"))
print([col["name"] for col in meta2.get("columns", [])])

200
Calgary Transit Realtime Vehicle Positions GTFS-RT
Calgary Transit realtime data updated every 30 seconds. Visit the Google Transit API page for more information on the <a href="https://developers.google.com/transit/gtfs-realtime/">GTFS-RT specification</a> and the <a href="https://developers.google.com/transit/gtfs-realtime/guides/feed-entities">feed entities</a> (<a href="https://data.calgary.ca/dataset/GTFS-RT-Trip-Updates/gs4m-mdc2/">Trip Updates</a>, <a href="https://data.calgary.ca/dataset/GTFS-RT-Alerts/jhgn-ynqj">Service Alerts</a> and Vehicle Positions). Also, see <a href="http://www.calgarytransit.com/service-updates?nid=170214">Service Updates</a>.
[]


In [15]:
params = {"$q": "transit", "$limit": 50}
response4 = requests.get(catalog_url, params=params)
results4 = response4.json()

print(f"Total found: {len(results4)}\n")
for item in results4:
    print(item.get("name"), "→", item.get("uid"), "| rows:", item.get("row_count"), "| cols:", item.get("column_count"))

Total found: 50

Calgary Equity Index - Drive to Work → sr9i-q3av | rows: None | cols: None
CEI Matrix - Diabetes Prevalence → m2q8-sk4i | rows: None | cols: None
CEI Matrix - Economic Opportunity Domain → am4g-kdk6 | rows: 452 | cols: 14
CEI Matrix - COPD Prevalence → e3fq-a3m7 | rows: None | cols: None
Calgary Transit Stops By Route Cross Reference → pm3p-838w | rows: 8352 | cols: 11
Calgary Transit Stops → muzh-c9qc | rows: 7748 | cols: 11
Calgary Transit Realtime Trip Updates GTFS-RT → gs4m-mdc2 | rows: None | cols: None
Calgary Transit Realtime Vehicle Positions GTFS-RT → am7c-qe3u | rows: None | cols: None
Calgary Equity Index - Lone Parent → am9x-e5t5 | rows: None | cols: None
Civic census 2011 - Travel by Transit to Work Map → yiwc-iv3x | rows: None | cols: None
Central Business District Cordon Count 2016 → dsje-mu8i | rows: 31 | cols: 18
Calgary Equity Index - 2024 → i6pu-a35d | rows: None | cols: None
Municipal Development Plan Primary Transit Network → mn7e-ps5j | rows: None

In [16]:
test_url2 = "https://data.calgary.ca/resource/64iy-rpqq.json"
test_response2 = requests.get(test_url2, params={"$limit": 5})
print(test_response2.status_code)
print(test_response2.text[:1500])

200
[{"year":"2007","month":"Apr","community_shuttle_operating_hours":"33128","ctrain_operating_hours":"11293","large_bus_operating_hours":"136040","date":"2007/04"}
,{"year":"2007","month":"Aug","community_shuttle_operating_hours":"36404","ctrain_operating_hours":"11838","large_bus_operating_hours":"144019","date":"2007/08"}
,{"year":"2007","month":"Dec","community_shuttle_operating_hours":"34360","ctrain_operating_hours":"11820","large_bus_operating_hours":"140081","date":"2007/12"}
,{"year":"2007","month":"Feb","community_shuttle_operating_hours":"30952","ctrain_operating_hours":"10474","large_bus_operating_hours":"136359","date":"2007/02"}
,{"year":"2007","month":"Jan","community_shuttle_operating_hours":"34204","ctrain_operating_hours":"11608","large_bus_operating_hours":"151694","date":"2007/01"}]



## Data Source Decision Log

**Original plan:** Use Calgary Transit Fleet List (ekzc-4ss8) as the primary vehicle dataset.

**Issue found:** This dataset returns `403 Forbidden` with the message 
"You must be logged in to access this resource" — confirmed via direct 
browser test, not a code or authentication header issue. The dataset's 
metadata catalog also no longer indexes it under searchable terms like 
"fleet" or "vehicle," suggesting it has been deprecated or restricted 
since its original publication.

**Resolution:** Queried Calgary's open data catalog (dataset 3fy4-eztt) 
to search for currently active, structured alternatives. Two candidates 
("Annual Corporate Energy Use by Bus Units" and "Realtime Vehicle 
Positions GTFS-RT") returned valid metadata but zero columns — likely 
deprecated or served via a non-standard format (GTFS-RT uses protobuf, 
not standard JSON).

**Final dataset lineup:**
- Calgary Transit Ridership (iema-jbc4) — demand data
- Calgary Transit Routes (hpnd-riq4) — network structure
- Hours of Service Provided by Calgary Transit (64iy-rpqq) — supply/utilization data

This substitution is arguably stronger than the original plan: combining 
ridership and operating hours allows direct calculation of utilization 
(boardings per operating hour), a core metric for fleet planning.